### PR TITLE
log: support --log-opt tag=

### PR DESCRIPTION
**Pull request by giuseppe**
support a custom tag to add to each log for the container.

Closes: https://github.com/containers/libpod/issues/3653

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: pkg/spec/createconfig.go [hunk 1]
```diff
@@ -287,6 +287,11 @@ func (c *CreateConfig) getContainerCreateOptions(runtime *libpod.Runtime, pod *l
 		options = append(options, libpod.WithLogPath(logPath))
 	}
 
+	logTag := getLoggingTag(c.LogDriverOpt)
```
Comment (vrothberg) [inline pkg/spec/createconfig.go]: "Logging" vs "Log" inconsistency

File: docs/source/markdown/podman-create.1.md [hunk 2]
```diff
@@ -419,6 +419,12 @@ Logging driver specific options.  Used to set the path to the container log file
 
 `--log-opt path=/var/log/container/mycontainer.json`
 
+**--log-opt**=*tag*
+
+Specify a custom log tag for the container.  For example:
+
+`--log-opt tag="{{.ImageName}}"`
```
Comment (vrothberg) [inline docs/source/markdown/podman-create.1.md]: What's the format? What are valid keys and values?

File: libpod/container_inspect.go [hunk 3]
```diff
@@ -630,16 +631,7 @@ type InspectNetworkSettings struct {
 }
 
 // Inspect a container for low-level information
```
Comment (vrothberg) [inline libpod/container_inspect.go]: Maybe extend the comment that this method expects callers to have locked the container (if needed)

**Comment by vrothberg** [comment]
https://review.rdoproject.org/zuul/buildset/fb366dee96094ad4bb02d1c41fb211b6 is failing in other PRs as well. Do you know what that is?

File: pkg/spec/createconfig.go (hunk 1 above)
Comment (giuseppe) [reply pkg/spec/createconfig.go]: I've used the same convention used above for `logPath := getLoggingPath(c.LogDriverOpt)`

File: docs/source/markdown/podman-create.1.md [hunk 4]
```diff
@@ -419,6 +419,16 @@ Logging driver specific options.  Used to set the path to the container log file
 
 `--log-opt path=/var/log/container/mycontainer.json`
 
+**--log-opt**=*tag*
+
+Specify a custom log tag for the container.  For example:
```
Comment (mheon) [inline docs/source/markdown/podman-create.1.md]: We should mention that other log options are available but not supported. Maybe "Set custom logging configuration. Presently supports the **tag** option which specified a custom log tag for the container. For example..."

File: pkg/spec/parse.go [hunk 5]
```diff
@@ -144,6 +144,18 @@ func getLoggingPath(opts []string) string {
 	return ""
 }
 
+func getLoggingTag(opts []string) string {
```
Comment (mheon) [inline pkg/spec/parse.go]: Can we combine this and getLoggingPath into one function, so we don't have to double-traverse the parameters? We can also add warnings for unrecognized options

**Comment by mheon** [comment]
I'm starting to really hate Go templates, but if we need to do it, we need to do it. Found a few small things, but overall looks good.

**Comment by TomSweeneyRedHat** [comment]
Tests aren't hip.
Other than the comments from @vrothberg and @mheon, LGTM

**Comment by rhatdan** [comment]
Is this something containers.conf should handle also?

File: libpod/container.go [hunk 6]
```diff
@@ -376,6 +376,8 @@ type ContainerConfig struct {
 	CgroupParent string `json:"cgroupParent"`
 	// LogPath log location
 	LogPath string `json:"logPath"`
+	// LogPath log location
```
Comment (NevilleC) [inline libpod/container.go]: this comment seems to be a duplicate of the `LogPath` one

**Comment by rhatdan** [comment]
@giuseppe could you respond to all of the questions above?

File: pkg/spec/parse.go (hunk 5 above)
Comment (giuseppe) [reply pkg/spec/parse.go]: done

File: docs/source/markdown/podman-create.1.md (hunk 2 above)
Comment (giuseppe) [reply docs/source/markdown/podman-create.1.md]: specified in the man page

**Comment by giuseppe** [comment]
> @giuseppe could you respond to all of the questions above?

I was still looking into adding tests.  We currently have all tests disabled for journald and checking whether the tag is correctly added will probably need to use `journalctl`.

> Is this something containers.conf should handle also?

since it is backend specific, should we have a way to specify `log_opt` instead?

**Comment by rhatdan** [comment]
We have LogSizeMax which maybe should be changed to log_opt,since this seems to be specific to file.

**Comment by giuseppe** [comment]
comments are addressed, but tests are missing

**Comment by giuseppe** [comment]
added a test but the journald tests are not enabled in the e2e tests.

I'll try to enable them as a follow up PR

**Comment by rhatdan** [comment]
What error will I see if I try to use log-opt without journald?

**Comment by giuseppe** [comment]
> What error will I see if I try to use log-opt without journald?

right now it will be silently ignored.  I think it should be fixed in conmon, I'll open a PR to address it

**Comment by giuseppe** [comment]
> right now it will be silently ignored. I think it should be fixed in conmon, I'll open a PR to address it

PR opened here: https://github.com/containers/conmon/pull/115

**Pull request merged**